Could not initialize cppcheck / Cannot invoke "org.eclipse.core.runtime.IPath.toFile()

Hello fellows,

I am so glad to see the comeback of “cppcheclipse”; I have missed it so much over the last few years.
Thank you for bring it back to live.

However I run immediately into the issue below. I am using the latest available version on Marketplace `1.2.1.v20250730-0905`

`07/02/2025, 00:07	Eclipse.org	Core Runtime	3.33.0.v20250206-0919	org.eclipse.core.runtime`

**Do you have an idea what could be a remedy for this exception?**

```
org.eclipse.core.runtime.CoreException: Could not initialize cppcheck for project PMP-ng
	at com.googlecode.cppcheclipse.ui.Builder$ResourceVisitor.initChecker(Builder.java:192)
	at com.googlecode.cppcheclipse.ui.Builder$ResourceVisitor.processFile(Builder.java:199)
	at com.googlecode.cppcheclipse.ui.Builder$ResourceVisitor.visit(Builder.java:122)
	at org.eclipse.core.internal.resources.Resource.lambda$1(Resource.java:154)
	at org.eclipse.core.internal.resources.Resource.lambda$0(Resource.java:122)
	at org.eclipse.core.internal.watson.ElementTreeIterator.doIteration(ElementTreeIterator.java:85)
	at org.eclipse.core.internal.watson.ElementTreeIterator.doIteration(ElementTreeIterator.java:90)
	at org.eclipse.core.internal.watson.ElementTreeIterator.doIteration(ElementTreeIterator.java:90)
	at org.eclipse.core.internal.watson.ElementTreeIterator.doIteration(ElementTreeIterator.java:90)
	at org.eclipse.core.internal.watson.ElementTreeIterator.iterate(ElementTreeIterator.java:135)
	at org.eclipse.core.internal.resources.Resource.accept(Resource.java:131)
	at org.eclipse.core.internal.resources.Resource.accept(Resource.java:92)
	at org.eclipse.core.internal.resources.Resource.accept(Resource.java:154)
	at org.eclipse.core.internal.resources.Resource.accept(Resource.java:142)
	at com.googlecode.cppcheclipse.ui.Builder.processResource(Builder.java:306)
	at com.googlecode.cppcheclipse.ui.commands.RunCodeAnalysis$RunCodeAnalysisJob.runResource(RunCodeAnalysis.java:26)
	at com.googlecode.cppcheclipse.ui.commands.ResourceSelectionJob.run(ResourceSelectionJob.java:44)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.core.runtime.IPath.toFile()" because the return value of "org.eclipse.cdt.core.settings.model.ACPathEntry.getLocation()" is null
	at com.googlecode.cppcheclipse.ui.ToolchainSettings.getIncludes(ToolchainSettings.java:182)
	at com.googlecode.cppcheclipse.ui.ToolchainSettings.getUserIncludes(ToolchainSettings.java:88)
	at com.googlecode.cppcheclipse.core.Checker.<init>(Checker.java:81)
	at com.googlecode.cppcheclipse.ui.Builder$ResourceVisitor.initChecker(Builder.java:164)
	... 17 more
```

--- a/com.googlecode.cppcheclipse.ui/src/com/googlecode/cppcheclipse/ui/ToolchainSettings.java
+++ b/com.googlecode.cppcheclipse.ui/src/com/googlecode/cppcheclipse/ui/ToolchainSettings.java
@@ -179,7 +179,12 @@ public class ToolchainSettings implements IToolchainSettings {
 					if (includePathSetting instanceof ACPathEntry) {
 						// let CDT do the resolution for all ACPathEntries which should cover all include settings
 						ACPathEntry includePathEntry = (ACPathEntry) includePathSetting;
-						includePath = includePathEntry.getLocation().toFile();
+						IPath location = includePathEntry.getLocation();
+					    if (location == null) {
+					        continue;
+					    }
+
+					    includePath = location.toFile();
 					} else {
 
 						// make path absolute

